src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,189 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddMethod(index);
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            if (head == null) {
+                tail = head = newNode;
+            } else {
+                newNode.next = head;
+                head.prev = newNode;
+                head = newNode;
+            }
+            size++;
+        } else if (index == size) {
+            newNode.prev = tail;
+            tail.next = newNode;
+            tail = newNode;
+            size++;
+        } else {
+            Node<T> current = head;
+            for (int i = 0; i < index - 1; i++) {
+                current = current.next;
+            }
+            current.next.prev = newNode;
+            newNode.next = current.next;
+            current.next = newNode;
+            newNode.prev = current;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T el : list) {
+                add(el);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        T tmp = current.value;
+        current.value = value;
+        return tmp;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = findNodeByIndex(index);
+        unlink(current);
+        return current.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = findIndexByValue(object);
+        if (index == -1) {
+            return false;
+        } else {
```

Comment:
```suggestion
        }
```
we don't need else after throwing/returning